esp/src/src-react/components/forms/CheckPermissions.tsx

```diff
@@ -0,0 +1,140 @@
+    refreshGrid?: () => void;
+    showForm: boolean;
+    setShowForm: (_: boolean) => void;
+}
+
+export const CheckPermissionsForm: React.FunctionComponent<CheckPermissionsFormProps> = ({
+    refreshGrid,
+    showForm,
+    setShowForm
+}) => {
+
+    const { handleSubmit, control, reset, setValue, watch } = useForm<CheckPermissionsFormValues>({ defaultValues });
+
+    const [showError, setShowError] = React.useState(false);
+    const [errorMessage] = React.useState("");
+    const [userOptions, setUserOptions] = React.useState<IDropdownOption[]>([]);
+    const [groupOptions, setGroupOptions] = React.useState<IDropdownOption[]>([]);
+    const selectedType = watch("SelectedType");
+
+    React.useEffect(() => {
+        const fetchUsersAndGroups = async () => {
+            try {
+                const users = Users();
+                const userOptions: IDropdownOption[] = Array.isArray(users) ? users.map(user => ({ key: user.id, text: user.name })) : [];
+                setUserOptions(userOptions);
+
+                const groups = Groups();
+                const groupOptions: IDropdownOption[] = Array.isArray(groups) ? groups.map(group => ({ key: group.id, text: group.name })) : [];
+                setGroupOptions(groupOptions);
+            } catch (error) {
+                logger.error("Error fetching users and groups:");
+            }
+        };
+
+        fetchUsersAndGroups();
+    }, []);
+
+    const closeForm = React.useCallback(() => {
+        setShowForm(false);
+    }, [setShowForm]);
+
+    const handleDropdownChange = (newValue: string, selectedType: string) => {
+        setValue("SelectedType", selectedType);
+        setValue("SelectedValue", newValue);
+    };
+
+    const onSubmit = React.useCallback((data) => {
+        const { FileName, SelectedType, SelectedValue } = data;
+        logger.info(`Checking permissions for file ${FileName} for ${SelectedType}: ${SelectedValue}`);
+    }, []);
+
+    return (
+        <MessageBox show={showForm} setShow={closeForm} title={nlsHPCC.CheckFilePermissions} minWidth={400}
+            footer={<>
+                <PrimaryButton text={nlsHPCC.Submit} onClick={handleSubmit(onSubmit)} />
+                <DefaultButton text={nlsHPCC.Cancel} onClick={() => { reset(defaultValues); closeForm(); }} />
+            </>}>
+            <Controller
+                control={control} name="FileName"
+                render={({
+                    field: { onChange, name: fieldName, value },
+                    fieldState: { error }
+                }) => <TextField
+                        name={fieldName}
+                        onChange={onChange}
+                        label={nlsHPCC.Name}
+                        value={value}
+                        errorMessage={error && error?.message}
+                    />}
+                rules={{
+                    required: nlsHPCC.ValidationErrorRequired
+                }}
+            />
+            <Dropdown
```

Comment:
I don't think this "Type" dropdown is required. It's not in the dojo version of that dialog at least.